kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -1312,7 +1333,11 @@ XmlComponentConfigurations loadEncryptedSnapshotFileContent(long snapshotID) thr
         }
 
         // File loaded, try to decrypt and unmarshall
-        String decryptedContent = new String(this.m_cryptoService.decryptAes(entireFile.toString().toCharArray()));
+        char[] decryptAes = this.m_cryptoService.decryptAes(entireFile.toString().toCharArray());
+        if (decryptAes == null) {
```

Comment:
This scenario covers case with a mocked service that returns null (i.e. a bad/default service implementation). There's no immediate danger for it happening here.